pkg/issuer/acme/dns/rfc2136/tsig_test.go

```diff
@@ -0,0 +1,159 @@
+// using the rfc2136 dynamic update.
+
+package rfc2136
+
+import (
+	"crypto/hmac"
+	"crypto/md5"
+	"crypto/sha1"
+	"crypto/sha256"
+	"crypto/sha512"
+	"encoding/base64"
+	"reflect"
+	"testing"
+
+	"github.com/miekg/dns"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_tsigHMACProvider_Generate(t *testing.T) {
+	var (
+		someSecret        = base64.StdEncoding.EncodeToString(([]byte("foo-secret")))
+		someMessage       = "foo-message"
+		someMessageMD5    = md5Message(t, []byte("foo-secret"), []byte("foo-message"))
+		someMessageSHA1   = sha1Message(t, []byte("foo-secret"), []byte("foo-message"))
+		someMessageSHA224 = sha224Message(t, []byte("foo-secret"), []byte("foo-message"))
+		someMessageSHA384 = sha384Message(t, []byte("foo-secret"), []byte("foo-message"))
+		someMessageSHA256 = sha256Message(t, []byte("foo-secret"), []byte("foo-message"))
+		someMessageSHA512 = sha512Message(t, []byte("foo-secret"), []byte("foo-message"))
+	)
+	tests := map[string]struct {
+		key     tsigHMACProvider
+		msg     []byte
+		tsig    *dns.TSIG
+		want    []byte
+		wantErr bool
+	}{
+		"message gets signed with HMACMD5": {
+			key:  tsigHMACProvider(someSecret),
+			msg:  []byte(someMessage),
+			tsig: &dns.TSIG{Algorithm: dns.HmacMD5},
+			want: someMessageMD5,
+		},
+		"message gets signed with HMACSHA1": {
+			key:  tsigHMACProvider(someSecret),
+			msg:  []byte(someMessage),
+			tsig: &dns.TSIG{Algorithm: dns.HmacSHA1},
+			want: someMessageSHA1,
+		},
+		"message gets signed with HMACSHA224": {
+			key:  tsigHMACProvider(someSecret),
+			msg:  []byte(someMessage),
+			tsig: &dns.TSIG{Algorithm: dns.HmacSHA224},
+			want: someMessageSHA224,
+		},
+		"message gets signed with HMACSHA384": {
+			key:  tsigHMACProvider(someSecret),
+			msg:  []byte(someMessage),
+			tsig: &dns.TSIG{Algorithm: dns.HmacSHA384},
+			want: someMessageSHA384,
+		},
+		"message gets signed with HMACSHA256": {
+			key:  tsigHMACProvider(someSecret),
+			msg:  []byte(someMessage),
+			tsig: &dns.TSIG{Algorithm: dns.HmacSHA256},
+			want: someMessageSHA256,
+		},
+		"message gets signed with HMAC512": {
+			key:  tsigHMACProvider(someSecret),
+			msg:  []byte(someMessage),
+			tsig: &dns.TSIG{Algorithm: dns.HmacSHA512},
+			want: someMessageSHA512,
+		},
+		"unknown algorithm value results in an error": {
+			key:     tsigHMACProvider(someSecret),
+			tsig:    &dns.TSIG{Algorithm: "xyz"},
+			wantErr: true,
+		},
+		"secret that isn't base64 results in an error": {
+			key:     tsigHMACProvider("foo"),
+			wantErr: true,
+		},
+	}
+	for name, scenario := range tests {
+		t.Run(name, func(t *testing.T) {
+			got, err := scenario.key.Generate(scenario.msg, scenario.tsig)
+			if (err != nil) != scenario.wantErr {
+				t.Errorf("tsigHMACProvider.Generate() error = %v, wantErr %v", err, scenario.wantErr)
+			}
+			if !reflect.DeepEqual(got, scenario.want) {
+				t.Errorf("tsigHMACProvider.Generate() = %v, want %v", got, scenario.want)
+			}
```

Comment:
Why use `reflect.DeepEqual` for comparing byte slices? Wouldn't `bytes.Equal` be more appropriate? 
And `assert.Equal` might be even better and give readable output if the bytes differ.

Reply:
Thanks, that's definitely a better way to do it 👍🏼